### PR TITLE
mirror: Set DESTDIR to empty when building libbpf

### PR DESCRIPTION
**Pull request by lavinyya**
If a value is assigned to the variable DESTDIR when compiling bpftool, this variable will be passed to libbpf's Makefile.

In this case, libbpf's header files will be installed in an unexpected directory, causing bpftool to be unable to find the header files.

when run this command
```bash
OUTPUT=/root/output_3rd/bpftool_build/ DESTDIR=/root/output_3rd/bpftool_root/ V=1 make -j8 install
```

Part of the output is as follows, we can see that the path is wrong
```bash
make[1]: 进入目录“/root/bpftool/libbpf/src”
mkdir -p /root/output_3rd/bpftool_build/libbpf/staticobjs
if [ ! -d '/root/output_3rd/bpftool_root//root/output_3rd/bpftool_build/libbpf/include/bpf' ]; then install -d -m 755 '/root/output_3rd/bpftool_root//root/output_3rd/bpftool_build/libbpf/include/bpf'; fi;
cc -I. -I../include -I../include/uapi -g -O2 -Werror -Wall -std=gnu89 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-unknown-warning-option -Wno-format-overflow   -c bpf.c -o /root/output_3rd/bpftool_build/libbpf/staticobjs/bpf.o
```

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ $(LIBBPF_OUTPUT) $(BOOTSTRAP_OUTPUT) $(LIBBPF_BOOTSTRAP_OUTPUT) $(LIBBPF_HDRS_DI
 
 $(LIBBPF): $(wildcard $(BPF_DIR)/*.[ch] $(BPF_DIR)/Makefile) | $(LIBBPF_OUTPUT)
 	$(Q)$(MAKE) -C $(BPF_DIR) OBJDIR=$(patsubst %/,%,$(LIBBPF_OUTPUT)) \
-		PREFIX=$(LIBBPF_DESTDIR:/=) $(LIBBPF) install_headers
+		PREFIX=$(LIBBPF_DESTDIR:/=) DESTDIR="" $(LIBBPF) install_headers
```
Comment (qmonnet) [inline src/Makefile:45]:
```suggestion
		DESTDIR="" PREFIX=$(LIBBPF_DESTDIR:/=) $(LIBBPF) install_headers
```

File: src/Makefile [hunk 2]
```diff
@@ -51,7 +51,7 @@ $(LIBBPF_INTERNAL_HDRS): $(LIBBPF_HDRS_DIR)/%.h: $(BPF_DIR)/%.h | $(LIBBPF_HDRS_
 $(LIBBPF_BOOTSTRAP): $(wildcard $(BPF_DIR)/*.[ch] $(BPF_DIR)/Makefile) | $(LIBBPF_BOOTSTRAP_OUTPUT)
 	$(Q)$(MAKE) -C $(BPF_DIR) OBJDIR=$(patsubst %/,%,$(LIBBPF_BOOTSTRAP_OUTPUT)) \
 		PREFIX=$(LIBBPF_BOOTSTRAP_DESTDIR:/=) \
-		ARCH= CROSS_COMPILE= CC="$(HOSTCC)" LD="$(HOSTLD)" AR="$(HOSTAR)" $@ install_headers
+		ARCH= CROSS_COMPILE= CC="$(HOSTCC)" LD="$(HOSTLD)" AR="$(HOSTAR)" DESTDIR="" $@ install_headers
```
Comment (qmonnet) [inline src/Makefile:54]: Can we move `DESTDIR=""` to the beginning of the line above, please?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks a lot! I was not aware of the issue, but I can reproduce and the fix looks correct. It doesn't affect the Makefile in the kernel repo, so this mirror is the right place to submit.

I just have a request, could you please move `DESTDIR` to the same locations and order (with regards to `PREFIX`) as they appear in [the Makefile from the kernel repo](https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/tools/bpf/bpftool/Makefile?h=v6.8#n45)? The two Makefile are mostly similar, and keeping them as close as possible helps with sync-ups.

**Comment by lavinyya** [comment]
> I just have a request, could you please move `DESTDIR` to the same locations and order (with regards to `PREFIX`) as they appear in [the Makefile from the kernel repo](https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/tools/bpf/bpftool/Makefile?h=v6.8#n45)? The two Makefile are mostly similar, and keeping them as close as possible helps with sync-ups.

I've modified this pr as per your suggestion

**Review by qmonnet — APPROVED** [review]
Thanks a lot!

**Pull request merged**